tree/dataframe/src/RLoopManager.cxx

```diff
@@ -474,7 +534,14 @@ void RLoopManager::RunTreeProcessorMT()
 /// Run event loop over one or multiple ROOT files, in sequence.
 void RLoopManager::RunTreeReader()
 {
+   if (fEndEntry == fStartEntry) // empty range => no work needed
+      return;
+
    TTreeReader r(fTree.get(), fTree->GetEntryList());
+
+   if (r.SetEntriesRange(fStartEntry, fEndEntry)) // returning > 0 indicates invalid fStartEntry
+      throw std::runtime_error("RLoopManager: fStartEntry cannot be larger than the number of entries.");
```

Comment:
I think the error can be improved (the user doesn't need to know that there is a data member called `fStartEntry`. Maybe something like
```
A range of entries was passed in the creation of the RDataFrame, but the starting entry is larger than the total number of entries in the dataset
```
and maybe use `std::logic_error`